redfish-core/lib/assembly.hpp

```diff
@@ -365,9 +408,106 @@ inline void setAssemblylocationIndicators(
+                }
+                if (openbmc)
+                {
+                    std::optional<bool> readytoremove;
+                    if (!json_util::readJson(*openbmc, asyncResp->res,
+                                             "ReadyToRemove", readytoremove))
+                    {
+                        return;
+                    }
+                    if (readytoremove.value() == true)
+                    {
+                        // Call systemd to stop ADCSensor
+                        crow::connections::systemBus->async_method_call(
+                            [asyncResp](const boost::system::error_code ec) {
+                                if (ec)
+                                {
+                                    BMCWEB_LOG_ERROR << "Failed to Stop ADCSensor:"
+                                                 << ec;
+                                    messages::internalError(asyncResp->res);
+                                    return;
+                                }
+                                messages::success(asyncResp->res);
+                            },
+                        "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+                        "org.freedesktop.systemd1.Manager", "StopUnit",
+                        "xyz.openbmc_project.adcsensor.service", "replace");
+                    }
+                    else if (readytoremove.value() == false)
+                    {
+                        // Call systemd to start ADCSensor
+                        crow::connections::systemBus->async_method_call(
+                            [asyncResp](const boost::system::error_code ec) {
+                                if (ec)
+                                {
+                                    BMCWEB_LOG_ERROR << "Failed to Start ADCSensor:"
+                                                 << ec;
+                                    messages::internalError(asyncResp->res);
+                                    return;
+                                }
+                                messages::success(asyncResp->res);
+                            },
+                        "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+                        "org.freedesktop.systemd1.Manager", "StartUnit",
+                        "xyz.openbmc_project.adcsensor.service", "replace");
+                    }
+                    else if (!readytoremove)
+                    {
+                        BMCWEB_LOG_ERROR << "Property Missing ";
+                        messages::propertyMissing(asyncResp->res, "ReadyToRemove");
+                        return;
+                    }
+                }
+                else
+                {
+                    BMCWEB_LOG_ERROR << "Property Missing ";
+                    messages::propertyMissing(asyncResp->res,
+                   "OpenBMC");
+                    return;
+                }  
+            }
+            else
```

Comment:
In bmcweb we typically do 

    if (!oem)
    {
                BMCWEB_LOG_ERROR << "Property Missing ";
                messages::propertyMissing(asyncResp->res,
                "Oem");
                return;
    }
    
    not the if/else 
    
    Oh well.